docs/antrea-network-policy.md

```diff
@@ -1501,6 +1549,107 @@ The following kubectl commands can be used to retrieve CG resources:
     kubectl get cg.crd.antrea.io
 ```
 
+## Group
+
+A Group CRD represents a different way for specifying how workloads are grouped
+together, and is conceptually similar to the ClusterGroup CRD. Users will be able
+to refer to Groups in Antrea NetworkPolicy objects instead of specifying Pods and
+Namespace selectors every time.
+
+### Group CRD
+
+Below are some example Group specs:
+
+```yaml
+# Group that selects all Pods labeled role: db
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-sel
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+---
+# Group that selects all Pods labeled role: db in Namespace labeled env: prod.
+# This Group cannot be used by Antrea NetworkPolicy appliedTo because of the namespaceSelector.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-with-namespace
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+  namespaceSelector:
+    matchLabels:
+      env: prod
+---
+# Group that selects ipBlocks 10.0.10.0/24.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-ip-block
+spec:
+  # IPBlocks cannot be set along with PodSelector, NamespaceSelector or serviceReference.
+  ipBlocks:
+    - cidr: 10.0.10.0/24
+---
+# Group that selects serviceReference test-service in the default Namespace.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-svc-ref
+spec:
+  # ServiceReference cannot be set along with PodSelector, NamespaceSelector or ipBlocks.
+  serviceReference:
+    name: test-service
+    namespace: default
+---
+# Group that has the previous Groups as childGroups.
+apiVersion: crd.antrea.io/v1alpha3
+kind: Group
+metadata:
+  name: test-grp-nested
+spec:
+  childGroups: [test-grp-sel, test-grp-ip-blocks, test-grp-svc-ref]
+```
+
+### Restrictions and Key differences from ClusterGroup
+
+Group has a similar spec with ClusterGroup. However, there are key differences and
+restrictions.
+
+- A Group can be set in an Antrea NetworkPolicy's `appliedTo` and `to`/`from` peers.
+  When set in the `appliedTo` field, it cannot include `namespaceSelector`, since
+  Antrea NetworkPolicy is Namespace scoped. For example, the
+  `test-grp-with-namespace` Group in the [sample](#group-crd) cannot be
+  used by Antrea NetworkPolicy `appliedTo`.
+- Antrea will not validate the referenced Group resources; if the convention is
+  violated in the Antrea NetworkPolicy's `appliedTo` section or for any of the rules'
+  `appliedTo`, then Antrea will report a condition of type `Realizable` in the
+  NetworkPolicy status with `False` status, `NetworkPolicyAppliedToUnsupportedGroup`
+  reason and a detailed message.
+- The `childGroup` only accepts strings, and they will be considered as names of
+  the Groups and looked for in the policy's own Namespace. For example, if
+  child Group `child-0` exists in `ns-2`, it should not be added as part of
+  `ns-1/parentGroup-0` which references child Group `ns-1/child-0`.
```

Comment:
I think we should remove "which references child Group `ns-1/child-0`". I do not feel it provides useful information.